test/include/seqan3/test/cereal.hpp

```diff
@@ -47,17 +47,18 @@ template <cereal_input_archive in_archive_t, cereal_output_archive out_archive_t
 //!\endcond
 void do_cerealisation(value_t && value)
 {
-    tmp_filename filename{"cereal_test"};
+    tmp_directory tmp{};
+    auto filename = tmp.path() / "cereal_test";
```

Comment:
Ok I looked into the issue. Correct me if I am wrong:

* you copied over the beaviour from tmp_filename to tmp_directory (in some other PR)
* now you changed every occurence of tmp_filename to use tmp_directory

Even though in this case I marked, it is clear that one only wants a tmp_filename.

so:
* What about the change in the issue about the tmp_filename? That it only get's a default constructor?
* Would it be better to use it then in appropriate places right away instead of changing everying to tmp_directory and then in some places back to tmp_filename?

Reply:
I don't understand the your first point `What about the change in the issue about the tmp_filename? That it only get's a default constructor?`

About the second part "Would it be better to use it then in appropriate places right away instead of changing everying to tmp_directory and then in some places back to tmp_filename?"
You are trying to say that we should keep a `tmp_filename` class that only deals with temporary files? Sure, we could do that.

Reply:
Replying from my phone so I don't have the compete overview right now. hope everything looks right.

I am pretty sure we wanted to keep a `tmp_filename()`too. Doesn't it say so in the issue?

```cpp
// I need just one temp file, I don't care about the name:
seqan3::test::tmp_filename file{};
// If I do care about the extension (e.g. I/O)
file.replace_extension(".fa");. // Like Filesystem::path
// File should be deleted on reconstruction

// If need a directory, multiple files or some other specific behaviour:
seqan3::test::tmp_directory dir{};
// Add other dirs and files
// Everything in the directory should be deleted on deconstruction
```
 While writing this I wonder what happens if I do this
```cpp
{
    seqan3::test::tmp_filename file{};
    std::ofstream ofs{file.get_path()}; // constructs a file;
    file.replace_extension(".txt");
} // Tmp file should be deleted on deconstruction but the filename is different from the constructed file. ?? 
```

Reply:
> I am pretty sure we wanted to keep a `tmp_filename()`too. Doesn't it say so in the issue?
> 
Ups 😅! I somehow missed that. Will change everything accordingly.

Reply:
> I am pretty sure we wanted to keep a `tmp_filename()`too. Doesn't it say so in the issue?

I think the resolution under that issue https://github.com/seqan/product_backlog/issues/281#issuecomment-766807049 says: 

> For purist reasons we only want `seqan3::test::tmp_directory` with the following features: 

That means no `seqan3::test::tmp_filename` any more. The future way of things should be to only offer a temporary directory where you can do whatever you want. It will warn you if you stepped outside that directory (to protect you from deleting `/tmp/XXXXXX_random_dir/../../`), if you forgot to delete a temporary file, and other safety features. But it will clean that directory no matter what after the destruction of `seqan3::test::tmp_filename`.

We discussed that a `seqan3::test::tmp_filename` makes no sense as you normally want a deterministic file name + extension, but `tmp_filename` implies that you would create a random file name in tmp, i.e. `/tmp/XXXXXXX_random_file_name`. Even if you can specify an extension, it won't be as flexible as the approach of getting the random-ness from a random tmp_directory, i.e. `/tmp/XXXXX_random_folder/` and putting a file in it with a `deterministic_file_name_with.extension`.

Thus, we decided against the `seqan3::test::tmp_filename` altogether in favour for the two-step approach `seqan3::test::tmp_directory`, by creating a random tmp directory first and doing whatever you want in there.